system_setup/common/helpers.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# Copyright 2015-2021 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import subprocess
+
+config_ref = {
+    "wsl": {
+        "automount": {
+            "enabled": "automount",
+            "mountfstab": "mountfstab",
+            "root": "custom_path",
+            "options": "custom_mount_opt",
+        },
+        "network": {
+            "generatehosts": "gen_host",
+            "generateresolvconf": "gen_resolvconf",
+        },
+        "interop": {
+            "enabled": "interop_enabled",
+            "appendwindowspath": "interop_appendwindowspath",
+        }
+    },
+    "ubuntu": {
+        "GUI": {
+            "theme": "gui_theme",
+            "followwintheme": "gui_followwintheme",
+        },
+        "Interop": {
+            "guiintegration": "legacy_gui",
+            "audiointegration": "legacy_audio",
+            "advancedipdetection": "adv_ip_detect",
+        },
+        "Motd": {
+            "wslnewsenabled": "wsl_motd_news",
+        }
+    }
+}
+
+
+def get_windows_locale():
+    try:
+        process = subprocess.Popen(["powershell.exe", "-NonInteractive",
+                                    "-NoProfile", "-Command",
+                                    "(Get-Culture).Name"],
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE)
+        stdout, _ = process.communicate()
+        if process.returncode:
+            return None
```

Comment:
We should probably write stderr in a warning log here in addition to skip it, like "can’t determine Windows locale, fallbacking to…"

Reply:
Fixed.

Reply:
Okay but the is a typo in the logging message and a ": " at the end without displaying a value. Please fix.